src/Repository/TransactionRepository.php

```diff
@@ -44,4 +44,24 @@ public function findAllSuccessfulTransactionByEmail(string $emailAddress): array
             ->getResult()
         ;
     }
+
+    /**
+     * Total amount in cents
+     */
+    public function getTotalAmountCurrentYearByEmail(string $email): int
+    {
+        return (int) $this->createQueryBuilder('transaction')
+            ->innerJoin('transaction.donation', 'donation')
+            ->select('SUM(donation.amount)')
+            ->where('donation.emailAddress = :email')
+            ->andWhere('transaction.payboxResultCode = :success_code')
+            ->andWhere('YEAR(transaction.payboxDateTime) = YEAR(NOW())')
```

Comment:
In general it's not a good idea to call a function for each row in table, in this case YEAR.
What about `date BETWEEN :first_day_of_year AND :end_date` ?

Reply:
Ok ;)